sequencer/src/main/java/net/consensys/linea/utils/PriorityThreadPoolExecutor.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package net.consensys.linea.utils;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.FutureTask;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.RunnableFuture;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+public class PriorityThreadPoolExecutor extends ThreadPoolExecutor {
+  public PriorityThreadPoolExecutor(
+      final int corePoolSize,
+      final int maximumPoolSize,
+      final long keepAliveTime,
+      final TimeUnit unit,
+      final ThreadFactory threadFactory) {
+    super(
+        corePoolSize,
+        maximumPoolSize,
+        keepAliveTime,
+        unit,
+        new PriorityBlockingQueue<>(),
```

Comment:
the default size of PriorityBlockingQueue is 11 - do we think that will be sufficient during a period of peak activity if we are throttling?

Reply:
`PriorityBlockingQueue` is unbounded so it will grow to accommodate more pending bundles if needed.
It is important to remember that the queue amount of task is always linked to the bundle pool size, since if a bundle is removed from the pool its related forward task will be removed as well, so there is no risk the queue will grow forever.